### PR TITLE
build(client-client-utils): use fluid-tsc

### PR DESCRIPTION
**Pull request by jason-ha**
+ import file extensions added by ts2esm
+ test ESM when possible (jest gitHash.spec.ts uses rewire which only functions for CJS)
- remove unused browser remap in telemetry-utils

Note: package is internal so has simple exports structure
  browser remap requires custom injected package.json as webpack applies browser remap from the context of export destination. (The CJS browser remap in root package.json are unused.)

### CODE REVIEW

File: packages/common/client-utils/package.json [hunk 1]
```diff
@@ -62,8 +92,8 @@
 		"test:coverage": "c8 npm test",
 		"test:jest": "jest",
 		"test:mocha": "npm run test:mocha:esm && npm run test:mocha:cjs",
-		"test:mocha:cjs": "mocha --recursive \"dist/test/mocha/**/*.spec.*js\" --exit",
-		"test:mocha:esm": "mocha --recursive \"lib/test/mocha/**/*.spec.*js\" --exit",
+		"test:mocha:cjs": "mocha --recursive \"dist/test/mocha/**/*.spec.*js\" --exit -r node_modules/@fluid-internal/mocha-test-setup",
+		"test:mocha:esm": "mocha --recursive \"lib/test/mocha/**/*.spec.*js\" --exit -r node_modules/@fluid-internal/mocha-test-setup",
```
Comment (jason-ha) [inline packages/common/client-utils/package.json:96]: @DLehenbauer, why do we want that?

File: packages/common/client-utils/package.json [hunk 2]
```diff
@@ -15,12 +15,42 @@
...
+			}
+		},
+		"./internal": {
+			"import": {
+				"types": "./lib/index.d.ts",
+				"default": "./lib/index.js"
+			},
+			"require": {
+				"types": "./dist/index.d.ts",
+				"default": "./dist/index.js"
```
Comment (jason-ha) [inline packages/common/client-utils/package.json:53]: @DLehenbauer, this package is named internal. I thought that meant that no one should be reaching in here. And therefore no need to have api exports like this.

File: packages/common/client-utils/tsconfig.json [hunk 3]
```diff
@@ -1,13 +1,10 @@
...
-	"exclude": ["src/test/**/*"],
+	"extends": ["../../../common/build/build-common/tsconfig.node16.json"],
 	"compilerOptions": {
 		"rootDir": "./src",
-		"outDir": "./dist",
+		"outDir": "./lib",
 		"baseUrl": ".",
 	},
+	"include": ["src/**/*"],
+	"exclude": ["src/test/**/*"],
```
Comment (DLehenbauer) [inline packages/common/client-utils/tsconfig.json:9]: FYI - I after looking at the results, I decided not to automatically move include/exclude to the end in the conversion script.

Most of our tsconfigs have these at the top, which feels most natural.

File: packages/common/client-utils/src/test/jest/tsconfig.cjs.json [hunk 4]
```diff
@@ -1,14 +1,14 @@
 {
-	"extends": ["../../../../../../common/build/build-common/tsconfig.test.json"],
-	"include": ["./**/*"],
-	"references": [
-		{
-			"path": "../../..",
-		},
-	],
+	"extends": ["../../../../../../common/build/build-common/tsconfig.test.node16.json"],
```
Comment (jason-ha) [inline packages/common/client-utils/src/test/jest/tsconfig.cjs.json:2]: @DLehenbauer, I noticed the surrounding `[]` was removed in the base tsconfig.json file, but present in all of these others. I think it is fine to leave them be to get this merged.

Comment (jason-ha) [reply packages/common/client-utils/src/test/jest/tsconfig.cjs.json:2]: Weel, needs another fix per test failures; so I will clean these up too.

File: fluidBuild.config.cjs [hunk 5]
```diff
@@ -165,6 +165,7 @@ module.exports = {
 			"docs/themes/thxvscode/layouts/",
 			"docs/themes/thxvscode/static/assets/",
 			"docs/tutorials/.*\\.tsx?",
+			"packages/common/client-utils/src/cjs/package.json",
```
Comment (sonalideshpandemsft) [inline fluidBuild.config.cjs:168]: nit: description as to why this is added :)

**Review by sonalideshpandemsft — APPROVED** [review]
LGTM 🥳

File: fluidBuild.config.cjs (hunk 5 above)
Comment (jason-ha) [reply fluidBuild.config.cjs:168]: I think up mean a comment in this file. The PR (commit) description does mention this file. I will follow on with an update for this. Good call.

Comment (sonalideshpandemsft) [reply fluidBuild.config.cjs:168]: oops...I meant a comment.

File: packages/common/client-utils/package.json [hunk 6]
```diff
@@ -59,8 +61,10 @@
 		"test": "npm run test:mocha && npm run test:jest",
 		"test:coverage": "c8 npm test",
 		"test:jest": "jest",
-		"test:mocha": "mocha  --recursive \"dist/test/mocha/**/*.spec.*js\" --exit",
-		"tsc": "tsc",
+		"test:mocha": "npm run test:mocha:esm && npm run test:mocha:cjs",
```
Comment (sonalideshpandemsft) [inline packages/common/client-utils/package.json:64]: nit: do we want to skip cjs test?

Comment (jason-ha) [reply packages/common/client-utils/package.json:64]: Not for this one. This packages covers complicated things and we should test CJS for it at least for now.

File: fluidBuild.config.cjs (hunk 5 above)
Comment (jason-ha) [reply fluidBuild.config.cjs:168]: #19895

**Pull request merged**